src/pages/fieldManagement/settings/defaultFieldComponents/relationshipComponents/Type.jsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import { get, omit } from 'lodash-es';
+import { v4 as uuid } from 'uuid';
+
+import InputAdornment from '@material-ui/core/InputAdornment';
+import IconButton from '@material-ui/core/IconButton';
+import NewChildIcon from '@material-ui/icons/AddCircle';
+
+import TextInput from '../../../../../components/inputs/TextInput';
+import DeleteButton from '../../../../../components/DeleteButton';
+import Role from './Role';
+
+function deleteType(relationships, typeGuid) {
+  return omit(relationships, [typeGuid]);
+}
+
+function updateType(relationships, typeGuid, newTypeLabel) {
+  const type = get(relationships, typeGuid, {});
+  return {
+    ...relationships,
+    [typeGuid]: {
```

Comment:
So, if relationships already has a property with key typeGuid, it will simply be overwritten by the subsequent
[typeGuid]: {...}
?
If so, that's very cool!

Reply:
that's it! with object spreading the items at the end of the statement overwrite the items at the beginning.